### ISSUE TITLE
bug: Fix ISR Calibration and Add submission.csv Export for Kaggle Titanic Competition

### ISSUE THREAD

**Issue by clduab11** [clduab11/titanic-kaggle#7]
_**Issues Found:**_

**ISR Calibration Failed:** The ISR value is 0.0304, which is well below the required threshold of T ≥ 1.5. This indicates the Inference Stability Ratio needs recalibration.

**Missing submission.csv Creation:** The notebook loads an existing submission.csv but doesn't actually create one from the model predictions. It only reads the gender_submission.csv sample file.

This notebook implements the Helios ML Framework for the Titanic survival prediction but has two critical issues preventing successful Kaggle submission:

ISR (Inference Stability Ratio) calibration failure - Current value: 0.0304, Required: T ≥ 1.5
Missing submission.csv generation - The notebook needs to output predictions in the correct format

**Current State:**
Framework: Helios ML with ISR governance and QMV monitoring
Best Model: LightGBM with 83.39% CV accuracy
All QMV scores: PASS (< 0.03 threshold)
19 engineered RLAD features implemented
Ensemble uses MoT voting across 5 models

**Required Fixes:**
1. ISR Recalibration
Current ISR value: 0.0304 (FAILED)
Required threshold: T ≥ 1.5
Likely issue: The ISR calculation needs adjustment for the dataset size or the threshold needs recalibration for binary classification on small datasets (891 training samples)
2. Add submission.csv Generation
After model training and predictions, add code to:

# Generate predictions on test set
test_predictions = best_model.predict(test_features)  # or ensemble.predict()

# Create submission DataFrame
submission = pd.DataFrame({
    'PassengerId': test['PassengerId'],
    'Survived': test_predictions
})

# Ensure predictions are integers (0 or 1)
submission['Survived'] = submission['Survived'].astype(int)

# Save to CSV
submission.to_csv('submission.csv', index=False)
print(f"Submission saved: {len(submission)} predictions")
print(f"Survival rate in predictions: {submission['Survived'].mean():.2%}")
Technical Context:
Models trained: LightGBM, XGBoost, Random Forest, Gradient Boosting, Logistic Regression
Feature engineering complete with 19 features
Cross-validation implemented
Ensemble voting strategy defined

**Acceptance Criteria:**
ISR metric passes threshold (T ≥ 1.5) or document why threshold adjustment is appropriate
submission.csv is generated with correct format: PassengerId, Survived (0/1 integers)
File can be directly submitted to Kaggle competition
Maintain current model performance (83%+ accuracy)

**Files to Modify:**
Main notebook (contains model training and evaluation)
Potentially src/ modules if ISR calculation is there
Add submission export section after model evaluation

**Additional Notes:**
Kaggle submission format: https://www.kaggle.com/c/titanic/overview/evaluation
Consider adding a validation check that submission has 418 rows (test set size)
Document the ISR recalibration approach for future reference

Please implement these fixes while maintaining the existing Helios ML Framework architecture and performance metrics.
